juriscraper/opinions/united_states/federal_bankruptcy/bap1.py

```diff
@@ -0,0 +1,134 @@
+        "USBC - District of Puerto Rico": "PR",
+        "USBC - District of Massachusetts (Boston)": "MB",
+        "USBC - District of Maine (Portland)": "EP",
+        "USBC - District of Rhode Island": "RI",
+        "Bankrupcty Court of ME - Bangor": "EB",
+        "Bankruptcy Court of MA - Boston": "MB",
+        "Bankruptcy Court of MA - Springfield": "MS",
+        "Bankruptcy Court of ME - Portland": "EP",
+        "Bankruptcy Court - Rhode Island": "RI",
+        "Bankruptcy Court - San Juan Puerto Rico": "PR",
+        "Bankruptcy Court of MA - Worcester": "MW",
+        "Bankruptcy Court - Ponce Puerto Rico": "PR",
+        "Bankruptcy Court of NH, Concord": "NH",
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.base_url = self.url = "https://www.bap1.uscourts.gov/bapopn"
+        self.court_id = self.__module__
+
+        # There are 29 pages as of development in Dec 2023 (source indexes from 0)
+        self.back_scrape_iterable = range(29)[::-1]
+
+    def _download(self, request_dict: Optional[Dict] = None) -> HtmlElement:
+        # For a normal periodic scraper, get the last page where most recent opinions are
+        if self.base_url == self.url:
+            self.html = super()._download()
+            self.url = self.html.xpath("//li/a/@href")[-1]
+
+        return super()._download()
+
+    def _download_backwards(self, page_number: int) -> None:
+        self.url = f"{self.base_url}?page={page_number}"
+
+    def _process_html(self) -> None:
+        # Most recent opinions are on the last rows of the table
+        for row in self.html.xpath("//tr[td]")[::-1]:
+            opinion_number = row.xpath("td[1]")[0].text_content().strip()
+
+            try:
+                status = self.get_status_from_opinion_number(opinion_number)
+            except SkipRowError:
+                continue
+
+            lower_court = row.xpath("td[4]/span")[0].text_content().strip()
+            docket_number = row.xpath("td[2]")[0].text_content().strip()
+
+            case = {
+                "status": status,
+                "url": row.xpath("td[1]/a/@href")[0],  # opinion url
+                "docket": self.build_full_docket_number(
+                    docket_number, lower_court
+                ),
+                # Pub Date
+                "date": row.xpath("td[3]")[0].text_content().strip(),
+                # short title
+                "name": row.xpath("td[4]")[0].text.strip(),
+                # district
+                "lower_court": lower_court,
+            }
+
+            self.cases.append(case)
+
+    def build_full_docket_number(
+        self, docket_number: str, lower_court: str
+    ) -> str:
+        """
+        Full docket number has the lower court abbreviation in it
+        For each unique lower court in the opinions available on the source,
+        the linked opinion PDF was opened and the abbrevation extracted
+        """
```

Comment:
We have a style for docstrings ..

The format is a variation of the reStructuredText (reST) or Sphinx documentation style. 

In reStructuredText format, the docstring below in `extract_from_text`

**Brief Description:** A short description of what the function does. For example in "Extract Date Filed from text."

Parameters and Return Description: Each parameter and the return value are described with a colon (:) followed by a tag (param, return), the parameter or return type (scraped_text in your example), and a brief description of it.
:param scraped_text: describes the parameter scraped_text with the explanation "Text of scraped content."
:return: describes what the function returns, in this case, "date filed."

if you need a longer example or explanation if can follow the short description at the start of the doc string

Reply:
Great, I will follow this convention. In time will document some of the base classes methods too